R/extract.R

```diff
@@ -596,11 +684,28 @@ extract.btergm <- function(model, level = 0.95, include.nobs = TRUE, ...) {
   return(tr)
 }
 
+#' \code{\link{extract}} method for \code{btergm} objects
+#'
+#' \code{\link{extract}} method for \code{btergm} objects. These objects are
+#' created by the \code{\link[btergm]{btergm}} function in the \pkg{btergm} package.
+#'
+#' @param level Confidence interval has a 0.95 probability of containing 
+#'    the population mean
+#' @inheritParams extract,lm-method
+#' @return A \linkS4class{texreg} object.
+#'
+#' @method extract btergm
+#' @aliases extract.btergm
+#' @family extract
+#' @seealso \link{extract}
+#' @author Philip Leifeld
+#'
+#' @export
```

Comment:
Have you checked if importing `confint` from `btergm` works? Probably not, but we could try.

Reply:
```
set.seed(1912)

library(xergm)
networks <- list()
for(i in 1:10){            # create 10 random networks with 10 actors
    mat <- matrix(rbinom(100, 1, .25), nrow = 10, ncol = 10)
    diag(mat) <- 0           # loops are excluded
    nw <- network(mat)       # create network object
    networks[[i]] <- nw      # add network to the list
}

covariates <- list()
for (i in 1:10) {          # create 10 matrices as covariate
    mat <- matrix(rnorm(100), nrow = 10, ncol = 10)
    covariates[[i]] <- mat   # add matrix to the list
}

m1 <- btergm(networks ~ edges + istar(2) +
                         edgecov(covariates), R = 100)
```
passing 
`screenreg(m1, level = 0.80)`

the output is:
```
=======================================
                         Model 1       
---------------------------------------
edges                     -0.96 *      
                         [-1.20; -0.68]
istar2                    -0.15 *      
                         [-0.29; -0.05]
edgecov.covariates[[i]]    0.02        
                         [-0.09;  0.11]
---------------------------------------
Num. obs.                900           
=======================================
* 0 outside the confidence interval.

```

passing 

`screenreg(m1, level = 0.95)`

the output is: 

```
=======================================
                         Model 1       
---------------------------------------
edges                     -0.96 *      
                         [-1.33; -0.55]
istar2                    -0.15 *      
                         [-0.35; -0.00]
edgecov.covariates[[i]]    0.02        
                         [-0.16;  0.20]
---------------------------------------
Num. obs.                900           
=======================================
* 0 outside the confidence interval.
```

Hence, if I correctly understood your question, it works!

Reply:
No, I meant adding `@importFrom btergm confint` or something like that and testing if it works. (Don't remember whether the package or function needs to be listed first, you'll figure it out...)

Reply:
I inserted it, BUT it gives me an error loading the package and I'm not understanding why

```
Loading texreg
Version:  1.36.24
Date:     2019-04-07
Author:   Philip Leifeld (University of Essex)

Please cite the JSS article in your publications -- see citation("texreg").
Warning messages:
1: replacing previous import ‘statnet.common::colMeans.mcmc.list’ by ‘ergm::colMeans.mcmc.list’ when loading ‘ergm.count’ 
2: replacing previous import ‘statnet.common::sweep.mcmc.list’ by ‘ergm::sweep.mcmc.list’ when loading ‘ergm.count’ 
3: replacing previous import ‘statnet.common::colMeans.mcmc.list’ by ‘ergm::colMeans.mcmc.list’ when loading ‘tergm’ 
4: replacing previous import ‘statnet.common::sweep.mcmc.list’ by ‘ergm::sweep.mcmc.list’ when loading ‘tergm’ 
5: replacing previous import ‘statnet.common::colMeans.mcmc.list’ by ‘ergm::colMeans.mcmc.list’ when loading ‘statnet’ 
6: replacing previous import ‘statnet.common::sweep.mcmc.list’ by ‘ergm::sweep.mcmc.list’ when loading ‘statnet’ 
7: replacing previous import ‘statnet.common::sweep.mcmc.list’ by ‘ergm::sweep.mcmc.list’ when loading ‘btergm’ 
8: replacing previous import ‘statnet.common::colMeans.mcmc.list’ by ‘ergm::colMeans.mcmc.list’ when loading ‘btergm’ 
```

Reply:
Let's not import those functions then if not necessary. Only import functions from base and stats etc. Unless R CMD check starts complaining, of course.